python_modules/dagster/dagster/_core/storage/cloud_storage_compute_log_manager.py

```diff
@@ -62,10 +95,31 @@ def cloud_storage_has_logs(
         """Returns whether the cloud storage contains logs for a given log key."""
 
     @abstractmethod
+    def _upload_file_obj(
+        self, data: IO[bytes], log_key: Sequence[str], io_type: ComputeIOType, partial=False
+    ):
+        pass
+
     def upload_to_cloud_storage(
         self, log_key: Sequence[str], io_type: ComputeIOType, partial: bool = False
     ) -> None:
         """Uploads the logs for a given log key from local storage to cloud storage."""
+        # We've already truncated
+        if (tuple(log_key), io_type) in self._truncated:
+            return
```

Comment:
can we add some logs to the place where it first decides to truncate and the place where it decides not to upload due to already being truncated, just for debugging purposes? ideally you could reverse engineer the sequence of events from the raw process logs